PolyEngine/Core/Src/Optional.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "Defines.hpp"
+
+namespace Poly {
+	template<typename V>
+	class Optional {
+	public:
+		using ValueType = V;
+	public:
+		Optional() : initialized(false), dummy{} {}
+		Optional(const V&  value) : initialized(true), value_storage(value) {}
+		Optional(      V&& value) : initialized(true), value_storage(std::move(value)) {}
+		~Optional() { if (this->HasValue()) { this->Value().~V(); } }
+
+	public:
```

Comment:
Sure it is! ;) I place them as separators and so I don't have to scroll far up to check for visibility of a given class member. I'm not strongly opposed to removing them, though.